collector/pressure_linux.go

```diff
@@ -93,13 +100,18 @@ func NewPressureStatsCollector(logger *slog.Logger) (Collector, error) {
 
 // Update calls procfs.NewPSIStatsForResource for the different resources and updates the values
 func (c *pressureStatsCollector) Update(ch chan<- prometheus.Metric) error {
+	foundResources := 0
 	for _, res := range psiResources {
 		c.logger.Debug("collecting statistics for resource", "resource", res)
 		vals, err := c.fs.PSIStatsForResource(res)
 		if err != nil {
-			if errors.Is(err, os.ErrNotExist) {
-				c.logger.Debug("pressure information is unavailable, you need a Linux kernel >= 4.20 and/or CONFIG_PSI enabled for your kernel")
-				return ErrNoData
+			if errors.Is(err, os.ErrNotExist) && res != psiResourceIRQ {
+				c.logger.Debug("pressure information is unavailable, you need a Linux kernel >= 4.20 and/or CONFIG_PSI enabled for your kernel", "resource", res)
+				continue
+			}
+			if errors.Is(err, os.ErrNotExist) && res == psiResourceIRQ {
+				c.logger.Debug("pressure information is unavailable, you need a Linux kernel >= 6.1 and/or CONFIG_PSI enabled for your kernel", "resource", res)
```

Comment:
A more distinct/specific error message may helps to understand the context. 

```suggestion
				c.logger.Debug("IRQ pressure information is unavailable, you need a Linux kernel >= 6.1 and/or CONFIG_PSI enabled for your kernel", "resource", res)
```

Reply:
Good idea. Fixed.